addons/l10n_it_edi/models/account_move.py

```diff
@@ -1031,117 +1029,84 @@ def _compose_info_message(self, tree, tags):
-                                " Origin Document Type field in the Electronic Invoicing tab."))
-            if self.l10n_it_origin_document_date and self.l10n_it_origin_document_date > fields.Date.today():
-                errors.append(_("The Origin Document Date cannot be in the future."))
-
-        errors += self._l10n_it_edi_export_taxes_data_check()
-
-        return errors
+    def _l10n_it_edi_update_header(self, errors):
+        """ Build each move's banner message """
+        move_errors_map = defaultdict(list)
+        if errors:
+            selectors = {
+                'company': lambda m, params: m.company_id.id in params[0],
+                'move':    lambda m, params: m.id.origin or m.id in params[0],
+                'partner': lambda m, params: m.commercial_partner_id.id in params[0],
+                'settings': lambda m, params: True,
+            }
+            for key, error in errors.items():
+                model_code = key.split("_")[0]
+                for move in self.filtered(lambda m: selectors[model_code](m, error['action_params'])):
+                    move_errors_map[move].append(error['message_single'])
+        for move in self:
+            (move._origin or move).l10n_it_edi_header = (
+                False if move not in move_errors_map
+                else self._l10n_it_edi_format_errors(None, move_errors_map[move])
+            )
 
-    def _l10n_it_edi_export_taxes_data_check(self):
-        """
-            Can be overridden by submodules like l10n_it_edi_withholding, which also allows for withholding and pension_fund taxes.
-        """
-        errors = []
-        for invoice_line in self.invoice_line_ids.filtered(lambda x: x.display_type == 'product'):
-            all_taxes = invoice_line.tax_ids.flatten_taxes_hierarchy()
-            vat_taxes = all_taxes.filtered(lambda t: t.amount_type == 'percent' and t.amount >= 0)
-            if len(vat_taxes) != 1:
-                errors.append(_("In line %s, you must select one and only one VAT tax.", invoice_line.name))
+    def _l10n_it_edi_base_export_check(self):
+        errors = {}
+        def build_error(message, message_single, records):
+            return {
+                'message': message,
+                'message_single': message_single,
+                'action_text': _("View invoices"),
+                'action_model': 'account.move',
+                'action_name': 'action_open_moves',
+                'action_params': [list(set(records.ids)), _("Invoices to check")],
+            }
+        if moves := self.filtered(lambda move: move.l10n_it_edi_is_self_invoice and move._l10n_it_edi_services_or_goods() == 'both'):
+            errors['move_reverse_charge_with_mixed_services_and_goods'] = build_error(
+                message=_("Cannot apply Reverse Charge to bills which contains both services and goods."),
+                message_single=_("Cannot apply Reverse Charge to a bill which contains both services and goods."),
```

Comment:
Is not it overkill to both provide a "single" and "multiple" message.  Mostly people accept "bill(s)"

Reply:
The "single" ends up on the move, the normal ends up in the wizard. We could do that in the code that reads the json, if there's no `message_single`, we take the `message` instead. Would that be ok?